### PR TITLE
Remove redundant rules

### PR DESCRIPTION
**Pull request by rplopes**
Remove set errors that are already included in `eslint:recommended` and unset errors that aren't included in that preset.

Also, replace deprecated rules by their corresponding alternatives.

Reference: https://eslint.org/docs/rules/

### CODE REVIEW

File: test/fixtures/correct.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const self = this;
 
 noop(self);
 
-// `curly` and `no-empty``.
+// `curly``.
```
Comment (waldyrious) [inline test/fixtures/correct.js]: You left out one extra backtick.

File: test/index.js [hunk 2]
```diff
@@ -34,19 +34,19 @@ describe('eslint-config-uphold', () => {
...
-      'no-empty',
+      'no-global-assign',
       'no-labels',
       'no-labels',
       'no-multi-str',
       'no-this-before-super',
       'no-underscore-dangle',
       'no-unused-vars',
       'padding-line-between-statements',
+      'padding-line-between-statements',
```
Comment (waldyrious) [inline test/index.js]: Duplicated line

Comment (rplopes) [reply test/index.js]: That's to be expected, as with some other rules in this test, since we're testing this rule with 2 different examples.

Comment (waldyrious) [reply test/index.js]: I see. Thanks for the clarification. 👍

File: src/index.js [hunk 3]
```diff
@@ -63,6 +62,7 @@ module.exports = {
     'no-eval': 'error',
     'no-extend-native': 'error',
     'no-extra-bind': 'error',
+    'no-global-assign': 'error',
```
Comment (Americas) [inline src/index.js]: This one is added by `eslint:recommended`

**Pull request merged**